routes/solution.go

```diff
@@ -70,6 +71,9 @@ func Solution(w http.ResponseWriter, r *http.Request) {
 
 	_, experimental := hole.ExperimentalByID[in.Hole]
 	if out.Pass && userID != 0 && !experimental {
+		// Log new records in Discord
+		discord.LogNewRecord(db, in.Code, session.Golfer(r).Name, in.Hole, in.Lang)
```

Comment:
Hmm this is before the solution is saved, I can see why you're doing it this way but I'd feel more comfortable if it happened after the transaction has committed, maybe `save_solution` could return all the info we need?